src/Orleans.Runtime/Hosting/EndpointOptions.cs

```diff
@@ -29,4 +39,65 @@ public class EndpointOptions
 
         //public bool BindToAny { get; set; }
     }
+
+    /// <summary>
+    /// Represent a network address with its mask
+    /// </summary>
+    public class NetworkAddress
+    {
+        /// <summary>
+        /// Address of the network
+        /// </summary>
+        public IPAddress Address { get; set; }
+
+        /// <summary>
+        /// Mask of the network
+        /// </summary>
+        public IPAddress Mask { get; set; }
+
+        public AddressFamily AddressFamily => Address.AddressFamily;
+
+        /// <summary>
+        /// Create a new NetworkAddress
+        /// </summary>
+        public NetworkAddress()
+            : this(IPAddress.Any, IPAddress.Any)
+        { }
+
+        /// <summary>
+        /// Create a new NetworkAddress
+        /// </summary>
+        public NetworkAddress(IPAddress address, IPAddress mask)
+        {
+            if (address.AddressFamily != mask.AddressFamily)
+                throw new ArgumentException("address and mask should have the same AddressFamily");
+            this.Address = address;
+            this.Mask = mask;
+        }
+
+        /// <summary>
+        /// Return true if <paramref name="otherAddress"/> belongs to this network
+        /// </summary>
+        internal bool Contains(IPAddress otherAddress)
+        {
+            if (this.Address.AddressFamily != otherAddress.AddressFamily)
+                return false;
+
+            var otherAddressBytes = otherAddress.GetAddressBytes();
+            var maskBytes = this.Mask.GetAddressBytes();
+
+            for (var i = 0; i < otherAddressBytes.Length; i++)
+            {
+                otherAddressBytes[i] = (byte)(otherAddressBytes[i] & maskBytes[i]);
+            }
+
+            return new IPAddress(otherAddressBytes).Equals(this.Address);
+        }
```

Comment:
This method only works if all bits which are 0 in `Mask` are also 0 in `Address`. i.e this will fail:
```C#
var netAddress = new NetworkAddress(IPAddress.Parse("10.0.0.1"), IPAddress.Parse("255.0.0.0"));
Assert.True(netAddress.Contains(IPAddress.Parse("10.0.0.2")));
```

So if I understand correctly, we should consider verifying that's true in the ctor.